solr/core/src/test/org/apache/solr/cloud/TestShardsInfoResponse.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.cloud;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.request.CollectionAdminRequest;
+import org.apache.solr.client.solrj.request.UpdateRequest;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.params.ShardParams;
+import org.apache.solr.common.util.SimpleOrderedMap;
+import org.hamcrest.MatcherAssert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.hamcrest.core.IsIterableContaining.hasItems;
+
+/**
+ * Test which asserts that shards.info=true works even if several shards are down
+ * It must return one unique key per shard.
+ * See SOLR-14892
+ */
+public class TestShardsInfoResponse extends SolrCloudTestCase {
+
+  @BeforeClass
+  public static void setupCluster() throws Exception {
+    configureCluster(3)
+        .addConfig("cloud-minimal", configset("cloud-minimal"))
+        .configure();
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void searchingWithShardsInfoMustNotReturnEmptyOrDuplicateKeys() throws Exception {
+
+    CollectionAdminRequest.createCollection("collection", "cloud-minimal", 3, 1)
+        .process(cluster.getSolrClient());
+
+    UpdateRequest update = new UpdateRequest();
+    for (int i = 0; i < 100; i++) {
+      update.add("id", Integer.toString(i));
+    }
+    update.commit(cluster.getSolrClient(), "collection");
+
+    // Stops 2 shards
```

Comment:
We have 3 nodes and 3 shards, but stopping 2 nodes does not necessarily means we stop 2 shards, depending on how shards are placed on nodes.

Any change this test might fail with some randomization?

Reply:
The failure came from `tidy`.
Just fixed it.

Reply:
My point was not about tidy. 🥲 
If we are unlucky and all shard are created on the same node, it seems to me this test will fail.
That's for sure if it can happen rare.

Reply:
The default algorithm will spread these out; not co-locate them.